File: src/test/java/de/flapdoodle/generics/local/j/HasLocalTypeMethodTest.java
package de.flapdoodle.generics.local.j;

import org.junit.jupiter.api.Test;

import java.util.function.Supplier;

import static org.junit.jupiter.api.Assertions.*;

class HasLocalTypeMethodTest {

  @Test
  void usage() {
    Factory<String, Supplier<String>> factory = x -> () -> x;
    String result = new HasLocalTypeMethod().outer(factory, "foo");
    assertEquals("foo", result);
  }
}

File: src/test/kotlin/de/flapdoodle/generics/local/k/HasLocalTypeMethodTest.kt
package de.flapdoodle.generics.local.k

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test
import java.util.function.Supplier

class HasLocalTypeMethodTest {

  @Test
  fun usage() {
    val factory = object : Factory<String, Supplier<String>> {
      override fun create(source: String): Supplier<String> {
        return Supplier { source }
      }
    }
    val result = HasLocalTypeMethod().outer(factory, "foo")
    assertEquals("foo", result)
  }

}